### PR TITLE
HTTP body size cap and `413` code logic.

### PR DESCRIPTION
**Pull request by dkorolev**

### CODE REVIEW

File: Blocks/HTTP/impl/posix_client.h [hunk 1]
```diff
@@ -101,9 +101,16 @@ class GenericHTTPClientPOSIX final {
         connection.BlockingWrite("Content-Type: " + request_body_content_type_ + "\r\n", true);
       }
       if (!request_body_contents_.empty()) {
-        connection.BlockingWrite("Content-Length: " + std::to_string(request_body_contents_.length()) + "\r\n", true);
-        connection.BlockingWrite("\r\n", true);
-        connection.BlockingWrite(request_body_contents_, false);
+        try {
```
Comment (nyospe) [inline Blocks/HTTP/impl/posix_client.h]: 😀

File: Bricks/net/http/impl/server.h [hunk 2]
```diff
@@ -78,8 +78,116 @@ inline static const http::Headers DefaultJSONHTTPHeaders() {
   return http::Headers({{"Access-Control-Allow-Origin", "*"}});
 }
 
+#ifndef CURRENT_MAX_HTTP_PAYLOAD
+constexpr const size_t kMaxHTTPPayloadSizeInBytes = 1024 * 1024 * 16;
+#else
+constexpr const size_t kMaxHTTPPayloadSizeInBytes = CURRENT_MAX_HTTP_PAYLOAD;
```
Comment (nyospe) [inline Bricks/net/http/impl/server.h]: nice!

File: Bricks/net/http/impl/server.h [hunk 3]
```diff
@@ -255,6 +362,11 @@ class GenericHTTPRequestData : public HELPER {
             HELPER::OnHeader(key, value);
             if (HeaderNameEquals(key, constants::kContentLengthHeaderKey)) {
               body_length = static_cast<size_t>(atoi(value));
+              if (body_length > constants::kMaxHTTPPayloadSizeInBytes) {
```
Comment (nyospe) [inline Bricks/net/http/impl/server.h]: Oh, very appropriate use of the exception here!

File: Bricks/net/http/impl/server.h [hunk 4]
```diff
@@ -273,13 +385,9 @@ class GenericHTTPRequestData : public HELPER {
...
               // Has HTTP body to parse.
               length_cap = body_offset + body_length;
-              // Resize the buffer to be able to get the contents of HTTP body without extra resizes,
-              // while being careful to not be open to extra-large mistakenly or maliciously set
-              // Content-Length.
               // Keep in mind that `buffer_` should have the size of `length_cap + 1`, to include the `\0'.
               if (length_cap + 1 > buffer_.size()) {
-                const size_t delta_size = length_cap + 1 - buffer_.size();
-                buffer_.resize(buffer_.size() + std::min(delta_size, buffer_max_growth_due_to_content_length));
+                buffer_.resize(length_cap + 1);
```
Comment (nyospe) [inline Bricks/net/http/impl/server.h]: 👍 KISS at its best.

File: Bricks/net/http/impl/server.h [hunk 5]
```diff
@@ -78,8 +78,116 @@ inline static const http::Headers DefaultJSONHTTPHeaders() {
...
 
+#ifndef CURRENT_MAX_HTTP_PAYLOAD
+constexpr const size_t kMaxHTTPPayloadSizeInBytes = 1024 * 1024 * 16;
+#else
+constexpr const size_t kMaxHTTPPayloadSizeInBytes = CURRENT_MAX_HTTP_PAYLOAD;
+#endif  // CURRENT_MAX_HTTP_PAYLOAD
+
 }  // namespace constants
 
+// HTTP response helpers. Used from both `GenericHTTPRequestData` and `GenericHTTPServerConnection`.
```
Comment (nyospe) [inline Bricks/net/http/impl/server.h]: Makes me wish github's diff utility had move detection...

File: Blocks/HTTP/test.cc [hunk 6]
```diff
@@ -926,3 +926,26 @@ TEST(HTTPAPI, ResponseSmokeTest) {
...
+                         .Register("/enough_is_enough",
+                                   [](Request r) {
+                                     ASSERT_FALSE(r.body.empty());
+                                     EXPECT_EQ("POST", r.method);
+                                     r("Fits.\n");
+                                   });
+
+  {
+    const auto response = HTTP(POST(Printf("http://localhost:%d/enough_is_enough", FLAGS_net_api_test_port),
+                                    std::string(16 * 1024 * 1024, '.')));
```
Comment (dkorolev) [inline Blocks/HTTP/test.cc]: @mzhurovich, :+1: , pushed this, and a broadened `catch` to see if it fixes the tests on Mac.

File: Blocks/HTTP/impl/posix_client.h [hunk 7]
```diff
@@ -105,9 +105,9 @@ class GenericHTTPClientPOSIX final {
           connection.BlockingWrite("Content-Length: " + std::to_string(request_body_contents_.length()) + "\r\n", true);
           connection.BlockingWrite("\r\n", true);
           connection.BlockingWrite(request_body_contents_, false);
-        } catch (const net::SocketCouldNotWriteEverythingException&) {
+        } catch (const net::SocketWriteException&) {
           if (request_body_contents_.length() <= net::constants::kMaxHTTPPayloadSizeInBytes) {
-            // Swallow the `SocketCouldNotWriteEverythingException` exception for huge payloads, which Current rejects.
+            // Swallow the `SocketWriteException` exception for huge payloads, which Current rejects.
```
Comment (mzhurovich) [inline Blocks/HTTP/impl/posix_client.h]: Ok, let's add a comment that this logic is wrong and it's a temporary hack.

**Pull request merged**